server/node_test.go

```diff
@@ -340,20 +326,75 @@ func compareStoreStatus(t *testing.T, node *Node, expectedNodeStatus *status.Nod
 	return nodeStatus
 }
 
-// TestNodeStatus verifies that the store scanner correctly updates the node's
-// status.
-func TestNodeStatus(t *testing.T) {
+// compareStoreStatus ensures that the actual store status for the passed in
+// store is updated correctly. It checks that the Desc.StoreID, Desc.Attrs,
+// Desc.Node, Desc.Capacity.Capacity, NodeID, RangeCount, ReplicatedRangeCount
+// are exactly correct and that the bytes and counts for Live, Key and Val are
+// at least the expected value.
+// The latest actual stats are returned.
+func compareStoreStatus(t *testing.T, ts *TestServer, store *storage.Store, expectedStoreStatus *storage.StoreStatus, testNumber int) *storage.StoreStatus {
+	// Retrieve store status from database.
+	storeStatusKey := keys.StoreStatusKey(int32(store.Ident.StoreID))
+	storeStatus := &storage.StoreStatus{}
+	if err := ts.db.GetProto(storeStatusKey, storeStatus); err != nil {
+		t.Fatalf("%v: failure getting store status: %s", testNumber, err)
+	}
+
+	// Values much match exactly.
+	if expectedStoreStatus.Desc.StoreID != storeStatus.Desc.StoreID {
+		t.Errorf("%v: actual Desc.StoreID does not match expected\nexpected: %+v\nactual: %v\n", testNumber, expectedStoreStatus, storeStatus)
```

Comment:
these errors are very hard to read because they dump the entire structure every time. Would you mind picking up https://github.com/tamird/cockroach/commit/c91a91e922109a350b9bf62b05abf05d48614cfe into here?